src/ODataHttpClient/Models/Request.cs

```diff
@@ -3,55 +3,77 @@
 using ODataHttpClient.Serializers;
 using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Net.Http;
 using System.Net.Http.Headers;
 using System.Text;
 
 namespace ODataHttpClient.Models
 {
-    public class Request : IRequest
+	public class Request : IRequest
     {
         internal const string DEFAULT_TYPE_KEY = "odata.type";
         private static Type stringType = typeof(string);
         public HttpMethod Method { get; private set; }
-        public string Uri { get; private set; }
+		public HttpRequestHeaders Headers { get; private set; }
```

Comment:
I think that Headers props is IDictionary<string, string> enough.

Reply:
If this type is HttpRequestMessage, a user need headers helper like src/ODataHttpClient.Tests/Factories/HeadersFactory.cs

I think it is not usiful.

Reply:
> I think that Headers props is IDictionary<string, string> enough.

OK, I have other idea.

* The prop's type is  HttpRequestHeaders
* But Request factory method's headers prop's type is IDictionary<string, string>

It keep symmetry between Request and Response.

Reply:
@iwate, I choosed the second option, HeadersFactory will create `HttpRequestHeaders` from dictionary. 

I've uncertainty according next changes :
- Remove IRequest#Headers getter interface => it's using for unit tests for headers verification and in client: https://github.com/marcnet80/ODataHttpClient/blob/add-headers/src/ODataHttpClient/ODataClient.cs#L84
I think we should leave this property.

- Remove IRequest#CreateMessage(HttpRequestHeaders headers = null) interface =>
Can I add overloading? Interface `IRequest` will look like that:
<pre>
public interface IRequest
{
   HttpRequestHeaders Headers { get; }
   HttpRequestMessage CreateMessage();
   HttpRequestMessage CreateMessage(HttpRequestHeaders headers);
}
</pre>
@iwate, what do you think?

Reply:
> Remove IRequest#Headers getter interface => it's using for unit tests and in client:

I think you can achieve by using cast without adding Headers to IRequest.
```
Assert.True(((Request)request).Headers.Contains(headerKey));
```

> Remove IRequest#CreateMessage(HttpRequestHeaders headers = null) interface =>
> Can I add overloading? Interface IRequest will look like that:

Umm, No you can not. Sorry.
Request is Builder Class for HttpRequestMessage. I do not accept CreateMessage with any option;

Reply:
@iwate, ok, removed.